zanata-client-commands/src/main/java/org/zanata/client/commands/init/SourceConfigPrompt.java

```diff
@@ -117,6 +125,24 @@ SourceConfigPrompt promptUser() throws Exception {
         pushOptions.setSrcDir(srcDir);
         pushOptions.setIncludes(includes);
         pushOptions.setExcludes(excludes);
+
+
+        // if project type is file, we need to ask file type in order to find source documents
+        if (File.name().equalsIgnoreCase(pushOptions.getProjectType())) {
+
+            console.blankLine();
+            console.printfln(Question, get("project.file.type.question"));
+            RestClientFactory clientFactory = getClientFactory(pushOptions);
+            FileResourceClient client =
+                    clientFactory.getFileResourceClient();
+            rawDocumentTypes = client.acceptedFileTypes();
+            // this answer is not persisted in zanata.xml so user will still need to type it when they do the actual push
+            console.printfln(Hint, PushOptionsImpl.fileTypeHelp);
+            console.printf(Question, get("file.type.prompt"));
+            String answer = console.expectAnyNotBlankAnswer();
+            ((PushOptionsImpl) pushOptions).setFileTypes(answer);
+        }
```

Comment:
This file is so big, I think we could split it into different files that each handle querying the user for one or a few related things. Not necessary right now, but we should consider it if we add anything else. It would make it much easier to see (and change) the overall flow of prompts.